dev/fattest.simplicity/src/componenttest/topology/database/container/LibertyServerUtil.java

```diff
@@ -0,0 +1,192 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package componenttest.topology.database.container;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import com.ibm.websphere.simplicity.config.ConfigElementList;
+import com.ibm.websphere.simplicity.config.DataSource;
+import com.ibm.websphere.simplicity.config.DataSourceProperties;
+import com.ibm.websphere.simplicity.config.DatabaseStore;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.config.Variable;
+import com.ibm.websphere.simplicity.config.dsprops.Properties;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Server specific liberty config utilities used by the fattest.database project
+ */
+public class LibertyServerUtil {
+    //Logging Constants
+    private static final Class<LibertyServerUtil> c = LibertyServerUtil.class;
+
+    //No objects should be created from this class
+    private LibertyServerUtil() {}
+
+    /**
+     * For use when attempting to use <b>database rotation</b>. <br>
+     *
+     * Retrieves database specific properties from the provided JdbcDatabaseContainer, such as;
+     * username, password, etc. <br>
+     *
+     * Using the ServerConfiguration API. Retrieves all &lt;dataSource&gt; elements and modifies
+     * those that have the <b>fat.modify=true</b> attribute set. <br>
+     *
+     * This will replace the datasource &lt;properties.derby.*... &gt; with the specific &lt;properties.db... &gt
+     * for the provided JdbcDatabaseContainer. <br>
+     *
+     * @see              com.ibm.websphere.simplicity.config.ServerConfiguration
+     *
+     * @param  serv      - LibertyServer server instance being used for this FAT suite.
+     * @param  cont      - JdbcDatabaseContainer instance being used for database connectivity.
+     *
+     * @throws Exception
+     */
+    public static void setupDatabaseProperties(LibertyServer serv, JdbcDatabaseContainer<?> cont) throws Exception {
```

Comment:
CloneNotSupportedException isn't thrown by serv.getServerConfiguration().clone() it seems.

Reply:
Please double check me on this, but don't both `setupDatabaseProperties` and `setupGenericProperties` make the exact same call to `serv.getServerConfiguration().clone()`?

But since you already have `Exception` listed as being thrown, that should also cover `CloneNotSupportedException` since it is a subclass of `Exception`.